src/core/pipelines/segment/segment_fetcher.ts

```diff
@@ -85,6 +91,31 @@ export default function createSegmentFetcher<T>(
 ) : ISegmentFetcher<T> {
   const segmentLoader = createSegmentLoader(transport[bufferType].loader, options);
   const segmentParser = transport[bufferType].parser as any; // deal with it
+  const warning$ = new Subject<ICustomError>();
```

Comment:
could we do it without using a subject?

Reply:
Tried to do it. Take a look at my last commit.